src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationService.java

```diff
@@ -49,25 +60,74 @@ public class NocNotificationService {
         CaseData caseDataNew = caseDetailsNew.getCaseData();
-        String partyName = NocNotificationHelper.getRespondentNameForNewSolicitor(changeRequest, caseDataNew);
         CaseData caseDataPrevious = caseDetailsPrevious.getCaseData();
-        if (!isClaimantNonSystemUser(caseDataPrevious)) {
-            sendClaimantEmail(caseDetailsPrevious, caseDetailsNew, partyName);
-        }
+        String partyName;
 
-        if (changeRequest.getOrganisationToRemove() != null) {
-            String previousOrgId = changeRequest.getOrganisationToRemove().getOrganisationID();
-            sendEmailToOldOrgAdmin(previousOrgId, caseDataPrevious);
+        if (caseRoleId.getValue().getCode().equals(ClaimantSolicitorRole.CLAIMANTSOLICITOR.getCaseRoleLabel())) {
+            // send claimant noc change email
+            partyName = caseDataPrevious.getClaimant();
+            handleClaimantNocEmails(caseDetailsNew, partyName);
+        } else {
+            // send respondent noc change email
+            handleRespondentNocEmails(caseDetailsPrevious, caseDetailsNew, changeRequest);
+            partyName = NocNotificationHelper.getRespondentNameForNewSolicitor(changeRequest, caseDataNew);
         }
 
-        String newOrgId = changeRequest.getOrganisationToAdd().getOrganisationID();
-        sendEmailToNewOrgAdmin(newOrgId, caseDetailsNew, partyName);
+        // send organisation noc change email
+        handleOrganisationNocEmails(caseDataPrevious, caseDetailsNew, changeRequest, partyName, currentUserEmail);
 
+        // send tribunal noc change email
         sendTribunalEmail(caseDataPrevious);
+    }
+
+    public void handleClaimantNocEmails(CaseDetails caseDetailsNew, String partyName) {
+        CaseData caseDataNew = caseDetailsNew.getCaseData();
+
+        // send respondent or respondent solicitor noc change email
+        caseDataNew.getRespondentCollection().forEach(resp -> {
+            Map<String, String> emailAddressesMap =
+                    getRespondentAndRespRepEmailAddressesMap(caseDataNew, resp);
+
+            emailAddressesMap.forEach((email, respondentId) -> {
+                String caseLink = StringUtils.isNotBlank(respondentId)
+                        ? emailService.getSyrCaseLink(caseDetailsNew.getCaseId(), respondentId)
+                        : emailService.getExuiCaseLink(caseDetailsNew.getCaseId());
+                emailService.sendEmail(claimantTemplateId, email,
+                        buildPersonalisationWithPartyName(caseDetailsNew, partyName, caseLink));
+            });
+        });
+
+        // send claimant noc change email
+        String claimantEmail = NotificationHelper.getEmailAddressForClaimant(caseDataNew);
+
+        if (isNullOrEmpty(claimantEmail)) {
+            log.warn("missing claimantEmail");
+            return;
+        }
+
+        Map<String, String> personalisation = buildRespondentPersonalisation(caseDetailsNew, partyName);
```

Comment:
Minor - just wondering if buildRespondentPersonalisation should be renamed as the context it's being used in here is for the claimant